source/type/pb_converter.cpp

```diff
@@ -1932,6 +1916,26 @@ void PbCreator::fill_crowfly_section(const type::EntryPoint& origin, const type:
     default:
         throw navitia::exception("Unhandled TransportCaracteristic value in pb_converter");
     }
+
+    section->set_begin_date_time(navitia::to_posix_timestamp(origin_time));
+    section->set_duration(crow_fly_duration.total_seconds());
+    if (crow_fly_duration.total_seconds() > 0) {
+        section->set_length(origin.coordinates.distance_to(destination.coordinates));
+        auto* new_coord = section->add_shape();
+        new_coord->set_lon(origin.coordinates.lon());
+        new_coord->set_lat(origin.coordinates.lat());
+        new_coord = section->add_shape();
+        new_coord->set_lon(destination.coordinates.lon());
+        new_coord->set_lat(destination.coordinates.lat());
+    } else {
+        section->set_length(0);
+        // For teleportation crow_fly (duration=0), the mode is always 'walking'
+        section->mutable_street_network()->set_mode(pbnavitia::Walking);
```

Comment:
This will do the job.
Just an idea (maybe it can raise warnings in some others mind): We sometimes get 0 durations while still using a real fallback... Probably very rare and probably not a big problem to have walking in those case?
I don't know if we can check easily something else like the "is free" matrix (but then it should be tested at a higher level I guess).

Reply:
I'll merge the PR as your point isn't blocking. But please, raise a PR if you think this would be usefull :)